Resources/Private/JavaScript/index.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import {RedirectList} from './components/RedirectList';
+import Redirect from './interfaces/Redirect';
+
+import '../Styles/styles.scss';
+
+// Declare interface for Neos backend API
+declare global {
+    interface Window {
+        Typo3Neos: {
+            I18n: {
+                translate: Function;
+                addObserver: Function;
+                initialized: boolean;
+            },
+            Notification: {
+                notice: Function;
+                ok: Function;
+                error: Function;
+                warning: Function;
+                info: Function;
+            },
+        };
+    }
+}
+
+(() => {
```

Comment:
Is there a reason why you wrap that in a iif? (Immediately Invoked Function)
Your variables wont get added to window.

Alternative for a seperate scope is:
```
{
  const someVar = 'foobar';
}
```